packages/jaeger-ui/src/model/find-trace-name.test.js

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { getTraceName } from './trace-viewer';
+
+describe('getTraceName', () => {
+  const firstSpanId = 'firstSpanId';
+  const secondSpanId = 'secondSpanId';
+  const thirdSpanId = 'thirdSpanId';
+  const remoteSpanId = 'remoteSpanId';
+
+  const serviceName = 'serviceName';
+  const operationName = 'operationName';
+
+  const spansWithNoRoots = [
+    {
+      spanID: firstSpanId,
+      startTime: 1583758690000,
+      references: [
+        {
+          spanID: secondSpanId,
+          refType: 'CHILD_OF',
+        },
+      ],
+    },
+    {
+      spanID: secondSpanId,
+      startTime: 1583758680000,
+      references: [
+        {
+          spanID: thirdSpanId,
+          refType: 'CHILD_OF',
+        },
+      ],
+    },
+    {
+      spanID: thirdSpanId,
+      startTime: 1583758670000,
+      references: [
+        {
+          spanID: firstSpanId,
+          refType: 'CHILD_OF',
+        },
+      ],
+    },
+  ];
+  const spansWithMultipleRoots = [
+    {
+      spanID: firstSpanId,
+      startTime: 1583758690000,
+      references: [
+        {
+          spanID: thirdSpanId,
+          refType: 'CHILD_OF',
+        },
+      ],
+    },
+    {
+      spanID: secondSpanId, // root span
+      startTime: 1583758680000,
+    },
+    {
+      spanID: thirdSpanId, // root span
+      startTime: 1583758670000,
+      operationName,
+      process: {
+        serviceName,
+      },
+      references: [
+        {
+          spanID: remoteSpanId,
```

Comment:
nit: this looks more like `missingSpanId` rather than `remote`.

Reply:
Yes, its' better, [fixed](https://github.com/jaegertracing/jaeger-ui/pull/541/commits/08d8dbb7e98f7cf739d1d291d8e507f42e2d5b71).